### PR TITLE
Enable modern event system and delete dead code

### PR DESCRIPTION
**Pull request by gaearon**
In https://github.com/facebook/react/pull/19228, I forked legacy event system folder for DOM and Native.
In this PR, I am removing the `enableModernEventSystem` flag altogether:

- In DOM fork, I am removing all its falsy branches and consider `enableModernEventSystem = true`.
- In Native, we remove all its truthy branches and consider `enableModernEventSystem = false`.

This enables us to remove a bunch of files and codepaths in DOM. There's more to do there but this is already fairly invasive so I will send the rest as follow-ups.

You should review without whitespace. https://github.com/facebook/react/pull/19230/files?diff=split&w=1

### CODE REVIEW

File: packages/react-dom/src/events/plugins/__tests__/ModernSimpleEventPlugin-test.js [hunk 1]
```diff
@@ -231,15 +231,14 @@ describe('SimpleEventPlugin', function() {
...
-
       ReactDOM = require('react-dom');
       Scheduler = require('scheduler');
     });
 
     // @gate experimental
     it('flushes pending interactive work before extracting event handler', () => {
       container = document.createElement('div');
-      const root = ReactDOM.unstable_createRoot(container);
+      const root = ReactDOM.createRoot(container);
```
Comment (gaearon) [inline packages/react-dom/src/events/plugins/__tests__/ModernSimpleEventPlugin-test.js:241]: I didn't change this one. Git just thinks I renamed it whereas I actually deleted old test.

Kind of confusing why it uses unprefixed name. I thought that doesn't work anymore.

File: packages/react-native-renderer/src/legacy-events/SyntheticEvent.js [hunk 2]
```diff
@@ -77,10 +76,8 @@ function SyntheticEvent(
   this.dispatchConfig = dispatchConfig;
   this._targetInst = targetInst;
   this.nativeEvent = nativeEvent;
-  if (!enableModernEventSystem) {
-    this._dispatchListeners = null;
-    this._dispatchInstances = null;
-  }
+  this._dispatchListeners = null;
```
Comment (trueadm) [inline packages/react-native-renderer/src/legacy-events/SyntheticEvent.js:79]: We don't need these. I think you got the condition the wrong way around here.

File: packages/react-native-renderer/src/legacy-events/SyntheticEvent.js [hunk 3]
```diff
@@ -157,79 +154,71 @@ Object.assign(SyntheticEvent.prototype, {
    * won't be added back into the pool.
    */
   persist: function() {
-    // Modern event system doesn't use pooling.
-    if (!enableModernEventSystem) {
-      this.isPersistent = functionThatReturnsTrue;
-    }
+    this.isPersistent = functionThatReturnsTrue;
```
Comment (trueadm) [inline packages/react-native-renderer/src/legacy-events/SyntheticEvent.js:157]: Same here, we don't need this line.

File: packages/react-native-renderer/src/legacy-events/SyntheticEvent.js [hunk 4]
```diff
@@ -157,79 +154,71 @@ Object.assign(SyntheticEvent.prototype, {
...
-        }
-      }
-      this.dispatchConfig = null;
-      this._targetInst = null;
-      this.nativeEvent = null;
-      this.isDefaultPrevented = functionThatReturnsFalse;
-      this.isPropagationStopped = functionThatReturnsFalse;
-      this._dispatchListeners = null;
-      this._dispatchInstances = null;
+    const Interface = this.constructor.Interface;
```
Comment (trueadm) [inline packages/react-native-renderer/src/legacy-events/SyntheticEvent.js:171]: Same here

File: packages/react-native-renderer/src/legacy-events/SyntheticEvent.js [hunk 5]
```diff
@@ -339,28 +325,21 @@ function createOrGetPooledEvent(
...
 
 function addEventPoolingTo(EventConstructor) {
   EventConstructor.getPooled = createOrGetPooledEvent;
-
-  // Modern event system doesn't use pooling.
-  if (!enableModernEventSystem) {
-    EventConstructor.eventPool = [];
-    EventConstructor.release = releasePooledEvent;
-  }
+  EventConstructor.eventPool = [];
```
Comment (trueadm) [inline packages/react-native-renderer/src/legacy-events/SyntheticEvent.js:341]: And here

File: packages/react-native-renderer/src/legacy-events/SyntheticEvent.js (hunk 2 above)
Comment (gaearon) [reply packages/react-native-renderer/src/legacy-events/SyntheticEvent.js:79]: These are in the native renderer. The native renderer uses the old event system, so we need these.

**Comment by gaearon** [comment]
Hmm. I'm not sure I follow. The native renderer uses the _legacy_ system, right? We don't yet remove pooling on RN either.

**Pull request merged**